### PR TITLE
Updates to the api  uri signing service.

### PR DESCRIPTION
**Pull request by jrushford**
This PR adds restful update and delete functionality and adds documention.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/543/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/497/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/544/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/545/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/498/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/499/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/546/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/500/

File: docs/source/development/traffic_ops_api/v12/deliveryservice.rst [hunk 1]
```diff
@@ -3163,3 +3163,199 @@ URL Sig Keys
...
+  | ``kty``             | string | this parameter repeats for each jwt key in the array and specifies the key type as defined in RFC 7516.                                 |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``k``               | string | this parameter repeats for each jwt key in the array and specifies the base64 encoded symmetric key see RFC 7516.                       |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+
+  **Response Example** ::
+
+  {
+    "Kabletown URI Authority": {
+      "renewal_kid": "Second Key",
```
Comment (dewrich) [inline docs/source/development/traffic_ops_api/v12/deliveryservice.rst]: renewal_kid should be renewalKid (camelCase for API fields)

File: docs/source/development/traffic_ops_api/v12/deliveryservice.rst [hunk 2]
```diff
@@ -3163,3 +3163,199 @@ URL Sig Keys
...
+  +-----------+----------+----------------------------------------+
+
+  **Response Properties**
+
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  |    Parameter        |  Type  |                                                               Description                                                               |
+  +=====================+========+=========================================================================================================================================+
+  | ``Issuer``          | string | a string describing the issuer of the URI signing object. Multiple URISigning objects may be returned in a response, see example        |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``renewal_kid``     | string | a string naming the jwt key used for renewals.                                                                                          |
```
Comment (dewrich) [inline docs/source/development/traffic_ops_api/v12/deliveryservice.rst]: renewal_kid should be renewalKid (camelCase for API fields)

File: docs/source/development/traffic_ops_api/v12/deliveryservice.rst [hunk 3]
```diff
@@ -3163,3 +3163,199 @@ URL Sig Keys
...
+  +-----------+----------+----------------------------------------+
+
+  **Request Properties**
+
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  |    Parameter        |  Type  |                                                               Description                                                               |
+  +=====================+========+=========================================================================================================================================+
+  | ``Issuer``          | string | a string describing the issuer of the URI signing object. Multiple URISigning objects may be returned in a response, see example        |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``renewal_kid``     | string | a string naming the jwt key used for renewals.                                                                                          |
```
Comment (dewrich) [inline docs/source/development/traffic_ops_api/v12/deliveryservice.rst]: renewal_kid should be renewalKid (camelCase for API fields)

File: docs/source/development/traffic_ops_api/v12/deliveryservice.rst [hunk 4]
```diff
@@ -3163,3 +3163,199 @@ URL Sig Keys
...
+  | ``kty``             | string | this parameter repeats for each jwt key in the array and specifies the key type as defined in RFC 7516.                                 |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+  | ``k``               | string | this parameter repeats for each jwt key in the array and specifies the base64 encoded symmetric key see RFC 7516.                       |
+  +---------------------+--------+-----------------------------------------------------------------------------------------------------------------------------------------+
+
+  **Request Example** ::
+
+  {
+    "Kabletown URI Authority": {
+      "renewal_kid": "Second Key",
```
Comment (dewrich) [inline docs/source/development/traffic_ops_api/v12/deliveryservice.rst]: renewal_kid should be renewalKid (camelCase for API fields)

File: traffic_ops/traffic_ops_golang/riak.go [hunk 5]
```diff
@@ -22,17 +41,169 @@ type URISignerKeyset struct {
...
+	opts := &riak.ClusterOptions{
+		Nodes: nodes,
+	}
+	cluster, err := riak.NewCluster(opts)
+
+	return cluster, err
+}
+
+// endpoint handler for fetching uri signing keys from riak
+func getUrisignkeysHandler(db *sqlx.DB, cfg Config) http.HandlerFunc {
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: getUrisignkeysHandler should be getURIsignkeysHandler, URI is an acronym and should be all caps

File: traffic_ops/traffic_ops_golang/riak.go [hunk 6]
```diff
@@ -22,17 +41,169 @@ type URISignerKeyset struct {
...
+		handleErr := tc.GetHandleErrorFunc(w, r)
+
+		ctx := r.Context()
+		pathParams, err := getPathParams(ctx)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		xmlId := pathParams["xml-id"]
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: Query Parameters are also camelCase

https://cwiki.apache.org/confluence/display/TC/API+Guidelines

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/570/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/571/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/526/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/527/

File: traffic_ops/traffic_ops_golang/riak.go [hunk 7]
```diff
@@ -15,25 +34,177 @@ import (
 )
 
 const RiakPort = 8087
-const CdnUriKeysBucket = "cdn_uri_sig_keys" // riak namespace for cdn uri signing keys.
+const CdnURIKeysBucket = "cdn_uri_sig_keys" // riak namespace for cdn uri signing keys.
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: sorry missed this one CdnURIsKeyBucket should be CDNURIKeysBucket (CDN is an acronym)

File: traffic_ops/traffic_ops_golang/riak.go [hunk 8]
```diff
@@ -15,25 +34,177 @@ import (
...
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+		defer func() {
+			if err := cluster.Stop(); err != nil {
+				log.Errorf("%v\n", err)
+			}
+		}()
+
+		ro, err := fetchObjectValues(xmlId, CdnURIKeysBucket, cluster)
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: CDN is an acronym

File: traffic_ops/traffic_ops_golang/riak.go [hunk 9]
```diff
@@ -66,6 +237,34 @@ func assignDeliveryServiceUriKeysHandler(db *sqlx.DB, cfg Config) http.HandlerFu
...
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: CDN is an acronym

File: traffic_ops/traffic_ops_golang/riak.go [hunk 10]
```diff
@@ -75,7 +274,7 @@ func assignDeliveryServiceUriKeysHandler(db *sqlx.DB, cfg Config) http.HandlerFu
 			Value:           []byte(data),
 		}
 
-		err = saveObject(obj, CdnUriKeysBucket, db, cfg)
+		err = saveObject(obj, CdnURIKeysBucket, cluster)
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/riak.go]: CDN is an acronym

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/573/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/529/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/575/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/531/

**Pull request merged**